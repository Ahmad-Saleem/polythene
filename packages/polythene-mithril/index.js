export * from "polythene-mithril-base";
export * from "polythene-mithril-button";
export * from "polythene-mithril-fab";
export * from "polythene-mithril-icon";
export * from "polythene-mithril-raised-button";
export * from "polythene-mithril-ripple";
export * from "polythene-mithril-shadow";
export * from "polythene-mithril-svg";
