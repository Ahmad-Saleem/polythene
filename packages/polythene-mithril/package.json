{
  "name": "polythene-mithril",
  "version": "1.0.0-rc.1",
  "description": "",
  "main": "dist/polythene-mithril.js",
  "module": "dist/polythene-mithril.mjs",
  "scripts": {
    "clean": "rm -rf dist && mkdir dist",
    "rollup": "../../node_modules/rollup/bin/rollup -c ../../scripts/rollup.umd.js && ../../node_modules/rollup/bin/rollup -c ../../scripts/rollup.es.js",
    "build": "npm run rollup"
  },
  "files": [
    "dist",
    "README.md"
  ],
  "dependencies": {
    "polythene-mithril-base": "1.0.0-rc.1",
    "polythene-mithril-button": "1.0.0-rc.1",
    "polythene-mithril-card": "1.0.0-rc.1",
    "polythene-mithril-checkbox": "1.0.0-rc.1",
    "polythene-mithril-dialog": "1.0.0-rc.1",
    "polythene-mithril-dialog-pane": "1.0.0-rc.1",
    "polythene-mithril-fab": "1.0.0-rc.1",
    "polythene-mithril-icon": "1.0.0-rc.1",
    "polythene-mithril-icon-button": "1.0.0-rc.1",
    "polythene-mithril-list": "1.0.0-rc.1",
    "polythene-mithril-list-tile": "1.0.0-rc.1",
    "polythene-mithril-menu": "1.0.0-rc.1",
    "polythene-mithril-radio-button": "1.0.0-rc.1",
    "polythene-mithril-raised-button": "1.0.0-rc.1",
    "polythene-mithril-ripple": "1.0.0-rc.1",
    "polythene-mithril-shadow": "1.0.0-rc.1",
    "polythene-mithril-svg": "1.0.0-rc.1",
    "polythene-mithril-switch": "1.0.0-rc.1",
    "polythene-mithril-tabs": "1.0.0-rc.1",
    "polythene-mithril-textfield": "1.0.0-rc.1",
    "polythene-mithril-toolbar": "1.0.0-rc.1"
  },
  "author": "Arthur Clemens <arthurclemens@gmail.com> (http://visiblearea.com)",
  "license": "MIT"
}

