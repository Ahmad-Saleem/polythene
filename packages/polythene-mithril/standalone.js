export * from "./index";
import { addLayoutStyles, addFastClick } from "polythene-utilities";
import { addTypography, addRoboto } from "polythene-material-design";

addTypography();
addRoboto();
addLayoutStyles();
addFastClick();
