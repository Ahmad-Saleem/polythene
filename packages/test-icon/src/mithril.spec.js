import { runSnapshots } from "../../../scripts/tests/mithril-snapshots";
import { renderer, Icon } from "polythene-mithril";
import specTests from "./spec-tests.js";
import mithrilTests from "./tests-mithril.js";

runSnapshots({
  tests: specTests({ Icon, renderer }).concat(mithrilTests),
  renderer
});
