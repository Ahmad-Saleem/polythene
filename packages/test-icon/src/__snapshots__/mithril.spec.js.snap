exports[`test Child node (trusted svg children) -- dark tone class (color set with style option) 1`] = `
"<div style=\"color: rgb(255, 255, 255);\" class=
\"pe-icon pe-icon--regular\">
  <div class=\"pe-svg\">
    <svg width=\"24\" height=\"24\" viewbox=\"0 0 24 24\">
      <path d=
      \"M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm4.24 16L12 15.45 7.77 18l1.12-4.81-3.73-3.23 4.92-.42L12 5l1.92 4.53 4.92.42-3.73 3.23L16.23 18z\">
      </path>
    </svg>
  </div>
</div>
"
`;

exports[`test Child node (trusted svg children) 1`] = `
"<div class=\"pe-icon pe-icon--regular\">
  <div class=\"pe-svg\">
    <svg width=\"24\" height=\"24\" viewbox=\"0 0 24 24\">
      <path d=
      \"M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm4.24 16L12 15.45 7.77 18l1.12-4.81-3.73-3.23 4.92-.42L12 5l1.92 4.53 4.92.42-3.73 3.23L16.23 18z\">
      </path>
    </svg>
  </div>
</div>
"
`;

exports[`test Dark tone class + light tone 1`] = `
"<div style=\"background: rgb(255, 255, 255);\">
  <div class=\"pe-icon pe-icon--regular pe-light-tone\">
    <div class=\"pe-svg\">
      <svg width=\"24\" height=\"24\" viewbox=\"0 0 24 24\">
        <path d=
        \"M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm4.24 16L12 15.45 7.77 18l1.12-4.81-3.73-3.23 4.92-.42L12 5l1.92 4.53 4.92.42-3.73 3.23L16.23 18z\">
        </path>
      </svg>
    </div>
  </div>
  <div class=
  \"pe-icon pe-icon--regular pe-light-tone tests-icon-themed-icon\">
    <div class=\"pe-svg\">
      <svg width=\"24\" height=\"24\" viewbox=\"0 0 24 24\">
        <path d=
        \"M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm4.24 16L12 15.45 7.77 18l1.12-4.81-3.73-3.23 4.92-.42L12 5l1.92 4.53 4.92.42-3.73 3.23L16.23 18z\">
        </path>
      </svg>
    </div>
  </div>
</div>
"
`;

exports[`test Dark tone class + light tone class 1`] = `
"<div style=\"background: rgb(255, 255, 255);\" class=\"pe-light-tone\">
  <div class=\"pe-icon pe-icon--regular\">
    <div class=\"pe-svg\">
      <svg width=\"24\" height=\"24\" viewbox=\"0 0 24 24\">
        <path d=
        \"M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm4.24 16L12 15.45 7.77 18l1.12-4.81-3.73-3.23 4.92-.42L12 5l1.92 4.53 4.92.42-3.73 3.23L16.23 18z\">
        </path>
      </svg>
    </div>
  </div>
  <div class=\"pe-icon pe-icon--regular tests-icon-themed-icon\">
    <div class=\"pe-svg\">
      <svg width=\"24\" height=\"24\" viewbox=\"0 0 24 24\">
        <path d=
        \"M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm4.24 16L12 15.45 7.77 18l1.12-4.81-3.73-3.23 4.92-.42L12 5l1.92 4.53 4.92.42-3.73 3.23L16.23 18z\">
        </path>
      </svg>
    </div>
  </div>
</div>
"
`;

exports[`test No options 1`] = `
"<div class=\"pe-icon pe-icon--regular\"></div>
"
`;

exports[`test Option: avatar (type large) 1`] = `
"<div class=\"pe-icon pe-icon--regular pe-icon--avatar\">
  <img src=
  \"http://arthurclemens.github.io/assets/polythene/examples/avatar-1.png\">
</div>
"
`;

exports[`test Option: class 1`] = `
"<div class=\"pe-icon pe-icon--regular class-x\"></div>
"
`;

exports[`test Option: className 1`] = `
"<div class=\"pe-icon pe-icon--regular className-x\"></div>
"
`;

exports[`test Option: content (trusted svg content) 1`] = `
"<div class=\"pe-icon pe-icon--regular\">
  <div class=\"pe-svg\">
    <svg width=\"24\" height=\"24\" viewbox=\"0 0 24 24\">
      <path d=
      \"M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm4.24 16L12 15.45 7.77 18l1.12-4.81-3.73-3.23 4.92-.42L12 5l1.92 4.53 4.92.42-3.73 3.23L16.23 18z\">
      </path>
    </svg>
  </div>
</div>
"
`;

exports[`test Option: content 1`] = `
"<div class=\"pe-icon pe-icon--regular\">
  <svg width=\"24\" height=\"24\" viewbox=\"0 0 24 24\">
    <path d=
    \"M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm4.24 16L12 15.45 7.77 18l1.12-4.81-3.73-3.23 4.92-.42L12 5l1.92 4.53 4.92.42-3.73 3.23L16.23 18z\">
    </path>
  </svg>
</div>
"
`;

exports[`test Option: element 1`] = `
"<a class=\"pe-icon pe-icon--regular\"></a>
"
`;

exports[`test Option: id 1`] = `
"<div id=\"id-x\" class=\"pe-icon pe-icon--regular\"></div>
"
`;

exports[`test Option: size 1`] = `
"<div style=\"display: flex;\">
  <div class=\"pe-icon pe-icon--small\">
    <div class=\"pe-svg\">
      <svg width=\"24\" height=\"24\" viewbox=\"0 0 24 24\">
        <path d=
        \"M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm4.24 16L12 15.45 7.77 18l1.12-4.81-3.73-3.23 4.92-.42L12 5l1.92 4.53 4.92.42-3.73 3.23L16.23 18z\">
        </path>
      </svg>
    </div>
  </div>
  <div class=\"pe-icon pe-icon--regular\">
    <div class=\"pe-svg\">
      <svg width=\"24\" height=\"24\" viewbox=\"0 0 24 24\">
        <path d=
        \"M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm4.24 16L12 15.45 7.77 18l1.12-4.81-3.73-3.23 4.92-.42L12 5l1.92 4.53 4.92.42-3.73 3.23L16.23 18z\">
        </path>
      </svg>
    </div>
  </div>
  <div class=\"pe-icon pe-icon--medium\">
    <div class=\"pe-svg\">
      <svg width=\"24\" height=\"24\" viewbox=\"0 0 24 24\">
        <path d=
        \"M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm4.24 16L12 15.45 7.77 18l1.12-4.81-3.73-3.23 4.92-.42L12 5l1.92 4.53 4.92.42-3.73 3.23L16.23 18z\">
        </path>
      </svg>
    </div>
  </div>
  <div class=\"pe-icon pe-icon--large\">
    <div class=\"pe-svg\">
      <svg width=\"24\" height=\"24\" viewbox=\"0 0 24 24\">
        <path d=
        \"M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm4.24 16L12 15.45 7.77 18l1.12-4.81-3.73-3.23 4.92-.42L12 5l1.92 4.53 4.92.42-3.73 3.23L16.23 18z\">
        </path>
      </svg>
    </div>
  </div>
</div>
"
`;

exports[`test Option: src (image file) 1`] = `
"<div class=\"pe-icon pe-icon--regular\">
  <img src=
  \"http://arthurclemens.github.io/assets/polythene/examples/avatar-1.png\">
</div>
"
`;

exports[`test Option: src (svg file) 1`] = `
"<div class=\"pe-icon pe-icon--regular\">
  <img src=
  \"http://arthurclemens.github.io/assets/polythene/examples/recycle.svg\">
</div>
"
`;

exports[`test Option: style 1`] = `
"<div style=\"color: rgb(239, 108, 0);\" class=
\"pe-icon pe-icon--regular\">
  <div class=\"pe-svg\">
    <svg width=\"24\" height=\"24\" viewbox=\"0 0 24 24\">
      <path d=
      \"M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm4.24 16L12 15.45 7.77 18l1.12-4.81-3.73-3.23 4.92-.42L12 5l1.92 4.53 4.92.42-3.73 3.23L16.23 18z\">
      </path>
    </svg>
  </div>
</div>
"
`;

exports[`test Themed (color and size) -- dark tone class 1`] = `
"<div class=\"pe-icon pe-icon--regular tests-icon-themed-icon\">
  <div class=\"pe-svg\">
    <svg width=\"24\" height=\"24\" viewbox=\"0 0 24 24\">
      <path d=
      \"M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm4.24 16L12 15.45 7.77 18l1.12-4.81-3.73-3.23 4.92-.42L12 5l1.92 4.53 4.92.42-3.73 3.23L16.23 18z\">
      </path>
    </svg>
  </div>
</div>
"
`;

exports[`test Themed (color and size) 1`] = `
"<div class=\"pe-icon pe-icon--regular tests-icon-themed-icon\">
  <div class=\"pe-svg\">
    <svg width=\"24\" height=\"24\" viewbox=\"0 0 24 24\">
      <path d=
      \"M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm4.24 16L12 15.45 7.77 18l1.12-4.81-3.73-3.23 4.92-.42L12 5l1.92 4.53 4.92.42-3.73 3.23L16.23 18z\">
      </path>
    </svg>
  </div>
</div>
"
`;
